### PR TITLE
Feat typed host

### PR DESCRIPTION
**Pull request by DhariniJeeva**
this pr splits host into static, azure and AWS components similar to host-catalog and host-set.  

Note: only view mode is supported in AWS and Azure host. create, view and edit functions are supported in static host

### CODE REVIEW

File: addons/core/translations/resources/en-us.yaml [hunk 1]
```diff
@@ -155,6 +155,9 @@ host:
     unknown: Unknown
     aws: AWS
     azure: Azure
+  form:
+    provider:
```
Comment (randallmorey) [inline addons/core/translations/resources/en-us.yaml:159]: Let's consolidate on a single `provider` translation (maybe from host catalog?).  It's the same across all resources in the feature.

**Review by randallmorey — CHANGES_REQUESTED** [review]

File: addons/core/translations/resources/en-us.yaml (hunk 1 above)
Comment (DhariniJeeva) [reply addons/core/translations/resources/en-us.yaml:159]: good idea! I will consolidate

Comment (DhariniJeeva) [reply addons/core/translations/resources/en-us.yaml]: translation consolidation is done and merged [here](https://github.com/hashicorp/boundary-ui/pull/955)

**Review by calcaide — APPROVED** [review]
LGTM

**Pull request merged**